js/index.js

```diff
@@ -1,7 +1,95 @@
+console.log("I'm ready!");
+
+// 1.1 Create a variable `hacker1` with the driver's name.
+
+// 1.2 Print `"The driver's name is XXXX"`.
+
+// 1.3 Create a variable `hacker2` with the navigator's name.
+
+// 1.4 Print `"The navigator's name is YYYY"`.
+
 // Iteration 1: Names and Input
+let hacker1 = "Matteo"
+let hacker2 = "Arnold"
+  console.log('The driver\'s name is ' + hacker1 + '.');
+  console.log('The driver\'s name is ' + hacker2 + '.');
+
 
+// 2.1. Depending on which name is longer, print:
+// - The driver has the longest name, it has XX characters. or
+// - It seems that the navigator has the longest name, it has XX characters. or
+// - Wow, you both have equally long names, XX characters!.
 
 // Iteration 2: Conditionals
 
+console.log(hacker1.length);
+console.log(hacker2.length);
+
+
+if (hacker1.length > hacker2.length){
+  console.log('The driver ' + hacker1 + 'has the longest name, it has ' + hacker1.lenght + 'characters')
+} else if (hacker2.length > hacker1.length){
+  console.log('It seems that the navigator ' + hacker2 + 'has the longest name, it has ' + hacker2.length + 'characters')
+} else {
+  console.log('Wow, you both have equally long names, ' + hacker1.length + ' characters!')
+}
+
+// 3.1 Print all the characters of the driver's name, separated by a space and in capitals i.e. "J O H N"
+
+// 3.2 Print all the characters of the navigator's name, in reverse order. i.e. "nhoJ"
+
+// 3.3 Depending on the lexicographic order of the strings, print:
+// - The driver's name goes first.
+// - Yo, the navigator goes first definitely.
+// - What?! You both have the same name?
+
+// Iteration 3: Loops
+
+
+for (i=0; i<hacker1.length; i++) {
+// let arr= [hacker1];
+  console.log(hacker1[i].toUpperCase() + "\n");
+}
+
+for (i=0; i<hacker2.length; i++) {
+// let arr= [hacker2];
+  console.log(hacker2[i].toUpperCase() + "\n");
+}
+
+// Reverse name?
+// hacker1 = hacker1.split("");
+// hacker1 = hacker1.reverse();
+// hacker1 = hacker1.join("");
+
```

Comment:
You can also use a for loop which read the string starting at the end. Like for( i = hacker1.length -1; i >=0; i++){console.log(hacker[i])}